.github/workflows/build_macos.yaml

```
@@ -9,14 +9,19 @@ jobs:
       matrix:
         config: [debug, release]
         compiler: [clang]
-        arch: [x86, arm64]
+        os: [13, 14, 15]
+        # arch: [x86, arm64]]
         # Check https://github.com/actions/runner-images#available-images
         include:
-          - arch: x86
-            os: 12
-          - arch: arm64
-            os: 14
-    name: MacOS ${{ matrix.config }} ${{ matrix.compiler }} ${{ matrix.arch }}
+          - os: 13
+            osname: macOS13 x86
+          # - os: 14-large
+            # osname: macOS14 x86
+          - os: 14
+            osname: macOS14 arm64
+          - os: 15
+            osname: macOS15 arm64
+    name: MacOS ${{ matrix.config }} ${{ matrix.compiler }} ${{ matrix.osname}}
```

```suggestion
    name: MacOS ${{ matrix.os }} ${{ matrix.config }} ${{ matrix.compiler }} ${{ matrix.arch }}
```

(or without arch if we remove the include block and get arch inside the runner)

---

well I chose this due to the attempt to use more runners.    
I did not realize that not all runners are provided for FOSS. However the different runners for one macOS version are distinguished by a suffix like -large or -xlarge this would require the os variable to be for example 13 (or 13-large) which is  a x86 runner or 13-xlarge which is a arm runner.

as this would be of no benefit in the Artifact name I chose to go with the osname to combine Arch and OS Version to be used for artifacts and Jobnames 

However as for FOSS the runners with the suffix are not currently available we could just skip this PR and use your branch instead

---

OK, I guess it can stay this way in case we ever can use the suffixed runners.